cmd/ztest/ztest.c

```diff
@@ -2660,6 +2662,66 @@ ztest_spa_create_destroy(ztest_ds_t *zd, uint64_t id)
 	(void) rw_unlock(&ztest_name_lock);
 }
 
+/*
+ * Start and then stop the MMP threads to ensure the startup and shutdown code
+ * works properly.  Actual protection and property-related code tested via ZTS.
+ */
+/* ARGSUSED */
+void
+ztest_mmp_enable_disable(ztest_ds_t *zd, uint64_t id)
+{
+	ztest_shared_opts_t *zo = &ztest_opts;
+	spa_t *spa = NULL;
+
+	if (zo->zo_mmp_test)
+		return;
+
+	mutex_enter(&spa_namespace_lock);
```

Comment:
Is `spa_namespace_lock` the correct lock to take here?  I also see `ztest_name_lock` in the ztest code and am not clear on the reason for the latter.